src/sentry/utils/data_filters.py

```diff
@@ -51,7 +51,7 @@ def is_valid_release(project, release):
     Verify that a release is not being filtered
     for the given project.
     """
-    release = release.lower()
+    release = six.text_type(release).lower()
```

Comment:
Change these to use `force_text` as I did in here: https://github.com/getsentry/sentry/commit/7aec8588f53ec2366b8ccd92ba4612ac82bf37d1

Then also move this below the project option check as I did in that commit.